File: src/frontend/types/task.ts
export type TaskPriority = "low" | "medium" | "high" | "urgent";
export type TaskStatus = "pending" | "in_progress" | "completed" | "cancelled";
export type TaskType =
  | "approve_estimate"
  | "confirm_availability"
  | "request_information"
  | "follow_up"
  | "customer_reactivation"
  | "send_reminder"
  | "review_message"
  | "other";

export interface Task {
  id: string;
  type: TaskType;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  assignedTo?: string;
  createdBy: string; // "ai" or user name
  createdAt: Date;
  dueDate?: Date;
  completedAt?: Date;
  completedBy?: string;
  relatedCustomerId?: string;
  relatedLeadId?: string;
  relatedEstimateId?: string;
  relatedConversationId?: string;
  metadata?: Record<string, unknown>;
}

export interface TaskFilter {
  status?: TaskStatus[];
  priority?: TaskPriority[];
  assignedTo?: string;
  type?: TaskType[];
  overdue?: boolean;
}

File: src/backend/schemas/task.py
from uuid import UUID
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class TaskBase(BaseModel):
    type: str
    title: str
    description: Optional[str] = None
    priority: str = "medium"
    assigned_to: Optional[str] = None
    due_date: Optional[datetime] = None
    related_customer_id: Optional[UUID] = None
    related_lead_id: Optional[UUID] = None
    related_estimate_id: Optional[UUID] = None
    related_conversation_id: Optional[UUID] = None


class TaskCreate(TaskBase):
    created_by: str


class TaskStatusUpdate(BaseModel):
    status: str
    completed_by: Optional[str] = None


class TaskRead(TaskBase):
    id: UUID
    status: str
    created_by: str
    completed_at: Optional[datetime] = None
    completed_by: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}
